src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -339,21 +357,65 @@ private function email(Document $provider): ?EmailAdapter
     {
         $credentials = $provider->getAttribute('credentials');
         return match ($provider->getAttribute('provider')) {
+            'mock' => new Mock('username', 'password'),
             'mailgun' => new Mailgun($credentials['apiKey'], $credentials['domain'], $credentials['isEuRegion']),
-            'sendgrid' => new SendGrid($credentials['apiKey']),
+            'sendgrid' => new Sendgrid($credentials['apiKey']),
             default => null
         };
     }
 
     private function buildEmailMessage(Document $message, Document $provider): Email
     {
-        $from = $provider['options']['from'];
+        $fromName = $provider['options']['fromName'];
+        $fromEmail = $provider['options']['fromEmail'];
+        $replyToEmail = null;
+        $replyToName = null;
+        $cc = null;
+        $bcc = null;
+
+        if (isset($provider['options']['replyToName']) && isset($provider['options']['replyToEmail'])) {
+            $replyToName = $provider['options']['replyToName'];
+            $replyToEmail = $provider['options']['replyToEmail'];
+        }
+
+        if (isset($provider['options']['cc'])) {
+            foreach ($provider['options']['cc'] as $ccEmail) {
+                if (is_array($cc)) {
+                    $cc[] = [
+                        'email' => $ccEmail,
+                    ];
+                } else {
+                    $cc = [
+                        [
+                            'email' => $ccEmail,
+                        ]
+                    ];
+                }
+            }
+        }
+
+        if (isset($provider['options']['bcc'])) {
+            foreach ($provider['options']['bcc'] as $bccEmail) {
+                if (is_array($bcc)) {
+                    $bcc[] = [
+                        'email' => $bccEmail,
+                    ];
+                } else {
+                    $bcc = [
+                        [
+                            'email' => $bccEmail,
+                        ]
+                    ];
+                }
+            }
+        }
+
         $to = $message['to'];
         $subject = $message['data']['subject'];
         $content = $message['data']['content'];
         $html = $message['data']['html'];
 
-        return new Email($to, $subject, $content, $from, null, $html);
+        return new Email($to, $subject, $content, $fromName, $fromEmail, $replyToName, $replyToEmail, $cc, $bcc, html: $html);
```

Comment:
Do we still need the `html` param name here?

Reply:
yea, there's an attachments param in between